config/src/config.rs

```diff
@@ -610,6 +613,18 @@ impl NodeConfig {
             .filter(|network| RoleType::Validator == (&network.role).into())
             .last()
     }
+
+    /// Returns true if network_config is for an upstream network
+    pub fn is_upstream_network(&self, network_config: &NetworkConfig) -> bool {
```

Comment:
maybe instead?
```
pub fn is_upstream_network(&self, network_config: &NetworkConfig) -> bool {
        self.state_sync
            .upstream_peers
            .upstream_peers
            .iter()
            .any(|peer_id| network_config.network_peers.peers.contains_key(peer_id))
   }
```
no cloning and hashmap lookup utilized instead of  vector